src/server/server.ts

```diff
@@ -242,6 +274,145 @@ export class CTProtoServer<AuthRequestPayload, AuthData, ApiRequest extends NewM
     }
   }
 
+  /**
+   * Process not-first buffer message.
+   *
+   * @param client - connected client
+   * @param message - accepted message
+   */
+  private async handleBufferMessage(client: Client<AuthData, ApiResponse, ApiUpdate>, message: Buffer): Promise<void> {
+    /**
+     * Parsing meta data from buffer message
+     */
+    const fileIdSlice = 10;
+    const chunkNumberOffset = 10;
+    const sizeOffset = 14;
+
+    const fileId = message.slice(0,fileIdSlice).toString();
+    const chunkNumber = (message.readInt32BE(chunkNumberOffset));
+    const size = (message.readInt32BE(sizeOffset));
+
+    /**
+     * Getting file data
+     */
+    const dataSlice = 18;
+    let data = message.slice(dataSlice, dataSlice+size);
+
+    /**
+     * Parsing payload message in buffer message
+     */
+    let strPayload = message.slice(dataSlice + size).toString();
+
+    const payload = JSON.parse(strPayload);
+
+    let file = this.uploadedFiles.find((req) => req.id === fileId);
+
+    /**
+     * Meta data of the first chunk includes additional information
+     */
+    if ( chunkNumber == 0 ) {
+
+      /**
+       * Create new file to upload or add some data
+       */
+      if (file) {
+
+        /**
+         * Push payload file data if object file already created
+         */
+        file.file[0] = data;
+        file.chunks = payload.chunks;
+        file.type = payload.type;
+        file.payload = payload.payload;
+      } else {
+
+        /**
+         * Create new file object
+         */
+        this.uploadedFiles.push( {id: fileId, file: [data], chunks: payload.chunks, payload: payload.payload, type: payload.type} );
+        file = this.uploadedFiles.find((req) => req.id === fileId);
+      }
+    } else {
+      if (file) {
+
+        /**
+         * Push file data
+         */
+        file.file.push(data);
+      } else {
+
+        /**
+         * Create new uploading file
+         */
+        this.uploadedFiles.push({ id: fileId, file: [] });
+        let file = this.uploadedFiles.find((req) => req.id === fileId);
+        file!.file[chunkNumber] = data;
+      }
+    }
+
+    if (file?.chunks) {
+
+      /**
+       * Calculate percent of uploading
+       */
+      const percent = file.file.filter(Boolean).length/file.chunks * 100;
+
+
+      /**
+       * Respond uploading info
+       */
+      client.respond(payload.id, { percent: Math.floor(percent)+'%', type: file?.type, fileId: fileId });
```

Comment:
i think, the `%` is redundant, because you should return clear data, not a presentation

Reply:
and move props to separate lines